### ISSUE TITLE
[Regression in v1.2.0] `nerdctl logs` exits silently for `nerdctl run` w/o `-d`

### ISSUE THREAD

**Issue by AkihiroSuda** [containerd/nerdctl#1946]
```bash
nerdctl run --name=foo alpine echo foo
nerdctl logs foo
```

v1.1.0 fails as expected (because `-d` is not specified and the logging driver is not enabled)
```
FATA[0000] open /home/suda/.local/share/nerdctl/1935db59/containers/default/ed96e51c4c5f9b3d64cae624cba6b33ff3b9db3ed3b0224cc4bb02d1cbfc1a5c/log-config.json: no such file or directory
```

The main branch  exits silently, with exit code 0.

Regression in c67b102cb9d94a7df2657e410f28872f64555edd

**Comment by fahedouch** [comment]
Because of this https://github.com/containerd/nerdctl/commit/c67b102cb9d94a7df2657e410f28872f64555edd#diff-464b1e1a904b2f79ca04e2202190403c19f87a135b6a9e1ef2e65fa52772a221R85
 
Condition should be replaced by  `if logURI != ""`

**Comment by AkihiroSuda** [comment]
> Because of this [c67b102#diff-464b1e1a904b2f79ca04e2202190403c19f87a135b6a9e1ef2e65fa52772a221R85](https://github.com/containerd/nerdctl/commit/c67b102cb9d94a7df2657e410f28872f64555edd#diff-464b1e1a904b2f79ca04e2202190403c19f87a135b6a9e1ef2e65fa52772a221R85)
> 
> Condition should be replaced by `if logURI != ""`

Thanks, would you like to open a PR?

**Comment by fahedouch** [comment]
> > Because of this [c67b102#diff-464b1e1a904b2f79ca04e2202190403c19f87a135b6a9e1ef2e65fa52772a221R85](https://github.com/containerd/nerdctl/commit/c67b102cb9d94a7df2657e410f28872f64555edd#diff-464b1e1a904b2f79ca04e2202190403c19f87a135b6a9e1ef2e65fa52772a221R85)
> > Condition should be replaced by `if logURI != ""`
> 
> Thanks, would you like to open a PR?

Sure

**Comment by fahedouch** [comment]
We have an other issue here 
`nerdctl run --name=foo alpine echo foo` is not printing `foo` to stdout. Looks like `nerdctl logs` is not triggered behind the scene. Is it normal ?

**Comment by AkihiroSuda** [comment]
> `nerdctl run --name=foo alpine echo foo` is not printing `foo` to stdout.

Can't repro on my side

**Comment by fahedouch** [comment]
> > `nerdctl run --name=foo alpine echo foo` is not printing `foo` to stdout.
> 
> Can't repro on my side

@AkihiroSuda  Yes  sorry, it is fine until this [fix](https://github.com/containerd/nerdctl/pull/1947#issuecomment-1407480160)

**Comment by AkihiroSuda** [comment]
The regression isn't critical, so we can defer fixing this to v1.3 or later.

**Comment by austinvazquez** [comment]
Hi @fahedouch, I had a look at the purposed fix in the draft PR above and it looked like you are attempting to build a feature similar to Docker's dual logging. For the purpose of this issue, if the goal is to restore previous behavior, then the change would be to restore the line here: https://github.com/containerd/nerdctl/commit/c67b102cb9d94a7df2657e410f28872f64555edd#diff-21f959b0a9d34cee85c5c25014fe6cef81d7c30c93c5294231cf6ea80f0f65c9L495

Currently an empty file is being created for containers run without `-d` so which causes the `os.Stat` to succeed on `nerdctl logs` walk.

Dual logging is a nice to have but is likely much more involved and can be covered by #1657.

**Comment by austinvazquez** [comment]
> restore the the above line means no log persistance for nerdctl run w/o -d, that means we lost logs when restart container

Not quite, log persistence is not supported for containers ran without -d. See [here](https://github.com/containerd/nerdctl/blob/main/pkg/taskutil/taskutil.go#L85). Also restart container uses detached so that case is covered. Reference [here](https://github.com/containerd/nerdctl/blob/main/pkg/containerutil/containerutil.go#L232).

**Comment by fahedouch** [comment]
@austinvazquez 

> Not quite, log persistence is not supported for containers ran without -d. 

You are right. it is a non complete fix introduced by [this commit](https://github.com/containerd/nerdctl/commit/c67b102cb9d94a7df2657e410f28872f64555edd). the log persistence need this [change](https://github.com/containerd/nerdctl/pull/1947/files#diff-464b1e1a904b2f79ca04e2202190403c19f87a135b6a9e1ef2e65fa52772a221R85) to be ok, but this change needs the dual log to satisfy CI tests. So we have two option:

- rollback [commit](https://github.com/containerd/nerdctl/commit/c67b102cb9d94a7df2657e410f28872f64555edd) and work on it as a new feature
- consider this bug as non critical and continue working on [the dual logging setup](https://github.com/containerd/nerdctl/pull/1947) to unblock the CI ( we may defer fixing this to v1.3 or later)

**Comment by AkihiroSuda** [comment]
No need to rollback the commit, this is a non-critical.

**Comment by austinvazquez** [comment]
Sounds good thanks folks @fahedouch @AkihiroSuda. Will focus efforts on dual logging design.

**Comment by apostasie** [comment]
Hey folks.
Wondering if this is still a problem here?
With nerdctl v2 what we get is:

```
$ nerdctl run --name=foo alpine echo foo
foo
$ nerdctl logs foo
foo
```

**Comment by AkihiroSuda** [comment]
Thanks, this seems closable now, and the current behavior corresponds to Docker
